### PR TITLE
Prominent links: authors' UX

### PR DESCRIPTION
**Pull request by mlbrgl**
see https://www.notion.so/owid/Improve-prominent-links-authoring-experience-copy-paste-from-Google-Docs-6dc963e803f34363b48d6f2147e74428

## Author facing features
- transform eligible HTML and text links into prominent links when pasting content (e.g. from Google Docs)
- thinner footprint in Wordpress editor
- retrieve prominent links' title and image from the database when none entered
- #1126 
- send content errors to dedicated Slack channel
- Deprecated: https://www.notion.so/owid/Automatic-prominent-links-1eafcce85953483d912b6e5f20b928ec

### Internal redirects resolver (server side)
Resolve prominent links internal redirects on the server side, mimicking 301/302 redirects in the browser (see #1148 for applying the resolver to all links)

**Limits**
So far resolving recursively between wordpress redirects and grapher redirects. When the resolver exhausts wordpress and grapher redirects, it moves onto explorer redirects (defined in explorer code) for one last potential redirect. This means a potential (although unlikely) explorer -> grapher redirect (defined in explorer code) will not restart the recursive search for this grapher url. We might want to support that use case in the future, but the added complexity didn't seem warranted at this point. Incidentally, it is possible to short-circuit explorer redirects (defined in explorer code) by adding the explorer redirect in the Wordpress admin. In that case, the recursive search works as expected (since it never leaves the realm of Wordpress + grapher redirects)._

### "Kitchen sink" e2e tests
`yarn testCypress` fires up the cypress UI, where `kitchenSink.js` tests can be run on multiple prominent links configurations.

**Limits**
Cypress tests require a running dev environment (and doesn't run in CI). `kitchenSink.wordpress.html` keeps a manual backup of that page (which will be on the live install but kept private). If you want to run the tests before refreshing you local DB, create a Wordpress page, paste the HTML in editor mode and publish it. The page slug needs to be `kitchen-sink` (I'll probably switch to the preview URL once the page is created on the live instance and the post ID is final).

### CODE REVIEW

**Review by MarcelGerber — APPROVED** [review]
I skimmed over this and skipped some parts because I'm not very familiar with the whole Wordpress/Site code, but all that I saw looked good.

There are quite some `// todo` comments in there, are they meant to be addressed before merge?

File: baker/siteRenderers.tsx
Comment (mlbrgl) [reply baker/siteRenderers.tsx]: if (A) was defined in explorer code, then (B) defined in wordpress wouldn't be resolved. My explanation wasn't very clear, I updated it, let me know if that makes more sense now.

**Comment by mlbrgl** [comment]
@MarcelGerber thanks for the review! I checked the `//todo`, they are not meant to be addressed now, they are merely reminders of possible refactors.

File: baker/siteRenderers.tsx
Comment (danielgavrilov) [inline baker/siteRenderers.tsx:659]: Wow – nice! I didn't realise we could do it like this.

File: baker/siteRenderers.tsx
Comment (danielgavrilov) [inline baker/siteRenderers.tsx:490]: Not sure if authors are happy with this if it contains query parameters? (given that we recently [got rid of social media thumbnails](https://github.com/owid/owid-grapher/issues/1086))

But something that we could easily change later, so wouldn't block this PR.

File: baker/redirects.ts
Comment (danielgavrilov) [inline baker/redirects.ts:140]: Is this definitely the Netlify behaviour? That if the target URL defines some query parameter, all incoming query parameters from the original URL are ignored? (Just asking, have no idea and haven't checked, but would really like to know.)

**Review by danielgavrilov — APPROVED** [review]
Very very nice @mlbrgl! Everything I thought of that could go wrong you addressed (and elegantly!)

So now the only things left that could go wrong are the things we haven't thought of 🥳

File: clientUtils/urls/Url.ts
Comment (mlbrgl) [reply clientUtils/urls/Url.ts]: Good point, thanks! Now testing against more restrictive regex.

File: baker/siteRenderers.tsx
Comment (mlbrgl) [reply baker/siteRenderers.tsx:490]: @JoeHasell Daniel rightly points out that some of the grapher prominent links thumbnails will be out of sync (given that query parameters, such as e.g. `?country=~USA` will be ignored). This is already the case now.
Our options:
- (1) replace any grapher thumbnail with the default grapher thumbnail introduced by @danielgavrilov 
- (2) leave as is

Given:
- the size of those thumbnails, where accuracy cannot be evaluated
- the upcoming sunsetting of the large prominent link variation

I'd weakly be in favour of keeping them as-is, as they still give an indication of what to expect without requiring a click. Thoughts?

File: baker/redirects.ts
Comment (mlbrgl) [reply baker/redirects.ts:140]: @danielgavrilov yes, that's what I notice.

Considering the following redirect with no query params on the target (/public-spending/):
`/social-spending/ /public-spending/ 301!`

Screenshot 2022-02-16 at 09 00 51

--> the query param on the source is passed through

Now with a query param on the target: 
`/grapher/child-mortality-by-country /grapher/child-mortality-around-the-world?country=KEN 301!`

Screenshot 2022-02-16 at 08 56 22

--> the query param on the source is neither passed through...

Screenshot 2022-02-16 at 09 00 39

--> ... nor merged

**Comment by mlbrgl** [comment]
thanks for the review @danielgavrilov!

File: baker/redirects.ts
Comment (danielgavrilov) [reply baker/redirects.ts:140]: Interesting, great to know, thanks!

File: baker/siteRenderers.tsx
Comment (JoeHasell) [reply baker/siteRenderers.tsx:490]: Yes I agree @mlbrgl. In the short-term, leaving as it is now is probably the better of those two options. Having the default image isn't as 'wrong' as it is in the social media thumbnail case, and it does convey at least some information (even just to us). 

Longer-term we can revisit this question again in the wider context of entry redesign.

**Pull request merged**
